src/client/Microsoft.Identity.Client/ManagedIdentity/AbstractManagedIdentity.cs

```diff
@@ -308,5 +308,52 @@ private static void CreateAndThrowException(string errorCode,
 
             throw exception;
         }
+
+        /// <summary>
+        /// Sets the claims and capabilities in the request.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="parameters"></param>
+        protected virtual void ApplyClaimsAndCapabilities(
```

Comment:
nit: You are changing the `managedIdentityRequest` from the outside, It seems like this method snd the one below can live in the `ManagedIdentityRequest` class. Then you can do `request.AddClaimsAndCapabilities(...)`

Reply:
updated